util/set/set.go

```diff
@@ -45,3 +61,16 @@ func (s Set[T]) Contains(e T) bool {
 
 // Len reports the number of items in s.
 func (s Set[T]) Len() int { return len(s) }
+
+// Equal returns true if other is equal to s.
```

Comment:
Nit: The conventional locution for a predicate is "X reports whether Y".

Reply:
maps.Equal